### PR TITLE
POWR-1792 Pattern Lab Badge Component

### PR DESCRIPTION
**Pull request by TashaBasalt**
Setting up pattern lab demos for badge, pill badge, and nav alert badge.

To Test-
- Pull down this branch and go to http://portlandor.lndo.site/pattern-lab/?p=viewall-components-badge
- See each badge variant represented

### CODE REVIEW

File: web/themes/custom/cloudy/src/components/badge/badge.twig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{% set isPill = isPill|default(false) %}
+{% set isNavAlert = isNavAlert|default(false) %}
+
+{% set classes = [
+  'cloudy-badge',
+  'cloudy-badge--' ~ type,
+  isPill ? 'cloudy-badge-pill',
```
Comment (joekarasek) [inline web/themes/custom/cloudy/src/components/badge/badge.twig]: lets treat pill as a modifier and use `cloudy-badge--pill`, you will need to update the scss as well.

File: web/themes/custom/cloudy/src/components/badge/badge.twig [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  isNavAlert ? '.cloudy-badge-nav-alert',
```
Comment (joekarasek) [inline web/themes/custom/cloudy/src/components/badge/badge.twig]: Same here, 'cloudy-badge--nav-alert'

**Comment by TashaBasalt** [comment]
> Looks good. I left a few comments around some BEM standards. I also want to make sure we address any fallback logic needed around the two fields 'type' and 'isNavAlert'. What happens if neither is supplied, and what happens if both are supplied? The twig template can do a little bit of logic and default setting to manage these two situations.

I added a default type, but left everything else as is. If neither is supplied it defaults to the base badge, if both are supplied it defaults to pill badge. I didn't add too many restrictions as the full scope of badge use doesn't seemed to be fleshed out completely yet.

**Comment by joshuami** [comment]
> Looks good. I left a few comments around some BEM standards. I also want to make sure we address any fallback logic needed around the two fields 'type' and 'isNavAlert'. What happens if neither is supplied, and what happens if both are supplied? The twig template can do a little bit of logic and default setting to manage these two situations.

@joekarasek , is this good to merge? The PR looks fine and does not touch Drupal. Let me know and I'll merge. I have some other merges going today, so please let me do the actual merge to master.

**Comment by joekarasek** [comment]
@joshuami This is good to go. Merge away.

**Comment by joshuami** [comment]
@joekarasek, can you resolve the requested change? I can't seem to do that for you. (I can override and merge, but I like seeing all the green check marks 😄 )

**Pull request merged**